core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -183,17 +184,18 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   }
 
   def updateLogStartOffsetFromRemoteTier(remoteLogStartOffset: Long): Unit = {
-    if (!remoteLogEnabled()) {
-      error("Ignoring the call as the remote log storage is disabled")
-      return
-    }
     maybeIncrementLogStartOffset(remoteLogStartOffset, LogStartOffsetIncrementReason.SegmentDeletion)
   }
 
   def remoteLogEnabled(): Boolean = {
     UnifiedLog.isRemoteLogEnabled(remoteStorageSystemEnable, config, topicPartition.topic())
   }
 
+  def remoteLogEnabledOrRetainPolicy(): Boolean = {
```

Comment:
This method will always return `true` for topics that are not enabled with tiered storage. Can we  add a doc for clarity?

Should we also check whether LSO == LLSO? If true and `remoteLogEnabled()` is false, then return the value as `false` instead.

Reply:
Good suggestion! If LSO == LLSO, even if `remoteLogEnabled()`, it doesn't make sense to read from remote storage

Reply:
Updated in https://github.com/apache/kafka/pull/16653/commits/2bc9b0bca64fb48c448fc7141af20480be00e230.